### PR TITLE
[Not-for-review] fresh DPG libaries for reference doc

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK

Autorest.csharp will generate comments for reference doc(see issue https://github.com/Azure/autorest.csharp/issues/2288) . So we change Autorest.csharp for this feature.( see PR https://github.com/Azure/autorest.csharp/pull/2305).

This PR is to show the output of all the PRs in the entire azure-sdk-for-net after the Autorest.csharp codegen changes.

### CODE REVIEW

**Comment by heaths** [comment]
Before marking "ready for review", could you update the description with a summary of the change so we know what we're reviewing? Or, if it'd be easier to link to a PR for the generator that would have more information, that works as well. It'd be great and helpful to know what we're reviewing. In this case, it seems like it's updating the generator to better handle polymorphic models. Is that correct?

**Comment by annelo-msft** [comment]
> Before marking "ready for review", could you update the description with a summary of the change so we know what we're reviewing?

@heaths, this PR shows the output of a proposed generator change on the azure-sdk-for-net repo.  @chunyu3, could you update the description with this information as well as a link to the generator change and the issue this is addressing to help @heaths and other library owners understand what they're being asked to review?

File: sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs [hunk 1]
```diff
@@ -47,19 +47,13 @@ public ConfidentialLedgerClient(Uri ledgerUri, TokenCredential credential) : thi
         /// <summary> The constitution is a script that assesses and applies proposals from consortium members. </summary>
         /// <param name="context"> The request context, which can override default behaviors on the request on a per-call basis. </param>
         /// <remarks>
+        /// Below is the JSON schema for the request and response payloads.
+        /// Additional information can be found in the service REST API documentation:
+        /// https://docs.microsoft.com/rest/api/ConfidentialLedger/ConfidentialLedger/GetConstitution
```
Comment (annelo-msft) [inline sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs:52]: I get a 404 for this link.

Comment (annelo-msft) [reply sdk/confidentialledger/Azure.Security.ConfidentialLedger/src/Generated/ConfidentialLedgerClient.cs:52]: @christothes, would you be able to work with the ConfidentialLedger team to update their swagger file with https://swagger.io/specification/v2/#externalDocumentationObject values that link to REST API docs?

**Comment by chunyu3** [comment]
> > Before marking "ready for review", could you update the description with a summary of the change so we know what we're reviewing?
> 
> @heaths, this PR shows the output of a proposed generator change on the azure-sdk-for-net repo. @chunyu3, would you please update the description with this information as well as a link to the generator change and the issue this is addressing to help @heaths and other library owners understand what they're being asked to review?

Updated the description with the feature information and the changes in autorest.csharp which will apply.

File: sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Generated/ApplicationDataClient.cs [hunk 2]
```diff
@@ -69,53 +69,45 @@ public ApplicationDataClient(Uri endpoint, TokenCredential credential, FarmBeats
         /// <exception cref="ArgumentNullException"> <paramref name="farmerId"/> or <paramref name="applicationDataId"/> is null. </exception>
         /// <exception cref="ArgumentException"> <paramref name="farmerId"/> or <paramref name="applicationDataId"/> is an empty string, and was expected to be non-empty. </exception>
         /// <remarks>
+        /// Below is the JSON schema for the request and response payloads.
+        /// Additional information can be found in the service REST API documentation:
+        /// https://docs.microsoft.com/rest/api/FarmBeats/ApplicationData/Get
```
[the remark this answers is not in the document]
Comment (chunyu3) [reply sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Generated/ApplicationDataClient.cs:74]: Hi @annelo-msft we will be able to generate the right rest-api-doc link ourselves. If service team does not provide the link in swagger, we will not display the link in the <Remarks> for GA. And we will support to generate the right link after GA, and I will file an issue to track this. @annelo-msft is it OK for GA?

[the remark this answers is not in the document]
Comment (annelo-msft) [reply sdk/agrifood/Azure.Verticals.AgriFood.Farming/src/Generated/ApplicationDataClient.cs]: Yes, once the DPG generator supports generating the REST API link in the comments from `externalDocumentationObject`, we will have satisfied the requirement for links to REST API docs sufficient to GA.

File: sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Generated/ConversationAnalysisClient.cs [hunk 3]
```diff
@@ -310,47 +298,31 @@ public virtual Response JobStatus(Guid jobId, bool? showStats = null, RequestCon
...
-        ///       innererror: InnerErrorModel
-        ///     }
-        ///   }
+        ///   ], # Required. The set of tasks to execute on the input conversation.
         /// }
         /// </code>
         /// 
         /// </remarks>
-        public virtual async Task<Operation<BinaryData>> SubmitJobAsync(WaitUntil waitUntil, RequestContent content, RequestContext context = null)
+        public virtual async Task<Operation> SubmitJobAsync(WaitUntil waitUntil, RequestContent content, RequestContext context = null)
```
Comment (heaths) [inline sdk/cognitivelanguage/Azure.AI.Language.Conversations/src/Generated/ConversationAnalysisClient.cs:325]: The documentation changes look great, but this change (and similar) are breaking. I'm opening another issue to track some work from the archboard per @annelo-msft's request and will include a detailed breakdown of CLU's LROs.

**Review by heaths — CHANGES_REQUESTED** [review]

**Comment by chunyu3** [comment]
close this PR since autorest.csharp has already be upgraded to 3.0.0-beta.20220622.2 in other PR.

**Pull request closed without merging**